packages/taiko-client/driver/chain_syncer/event/blocks_inserter/pacaya.go

```diff
@@ -99,6 +102,11 @@ func (i *BlocksInserterPacaya) InsertBlocks(
 		lastPayloadData *engine.ExecutableData
 	)
 
+	// update lastBlockID
+	if i.latestBlockIDSeenInEventCh != nil {
+		i.latestBlockIDSeenInEventCh <- meta.GetLastBlockID()
```

Comment:
Since its sent / updated before the real unpreconf blocks / `headL1Origin` inserted into the L2 node, will there be an issue?

Reply:
But every Event *will* actually be processed, or else the chain will halt regardless, so we dont ever want to preconf state behind `meta.GetLastBlockID()` of the latest Event. Right? If there is an error in the event insertion, we still *dont want to preconf state behind meta.GetLastBlockID()`, there is an error in the driver itself that needs to recover or bug fix to process the events, and without this, `HeadL1Origin` doesnt update, or updates later (because it takes some time to process each block), and the sequencer can call `/preconfBlocks` and preconf bad state?